hpsklearn/estimator/_cost_fn.py

```diff
@@ -104,6 +104,8 @@ def _cost_fn(argd,
             Use multiple CPU cores when training estimators which support
             multiprocessing.
     """
+    # scikit-learn needs a legacy `numpy.random.RandomState` RNG
+    random_state = np.random.RandomState(random_state.bit_generator)
```

Comment:
the sklearn components that use this also accept an integer instead of a `RandomState`, so to maintain backwards compatibility the simplest thing to do would be to check if `random_state` is a `np.random.Generator` and only do the conversion if it is.

Reply:
Good catch! I've pushed the fixes.